### PR TITLE
78680 more unit tests

### PR DESCRIPTION
**Pull request by janid1967**
Added some missing tests, detecting when working with IPO/PO

### CODE REVIEW

File: src/tests/Equinor.Procosys.Preservation.MainApi.Tests/Permission/MainApiPermissionServiceTests.cs [hunk 1]
```diff
@@ -66,6 +66,44 @@ public async Task GetPermissions_ShouldReturnNoPermissions_OnInValidPlant()
             Assert.AreEqual(0, result.Count);
         }
 
+        [TestMethod]
+        public async Task GetAllProjectsAsync_ShouldReturnThreeProjects_OnValidPlant()
```
Comment (Cnordbo) [inline src/tests/Equinor.Procosys.Preservation.MainApi.Tests/Permission/MainApiPermissionServiceTests.cs:70]: "Should return three projects", but assertion is on "2"

File: src/tests/Equinor.Procosys.Preservation.MainApi.Tests/Permission/MainApiPermissionServiceTests.cs [hunk 2]
```diff
@@ -66,6 +66,44 @@ public async Task GetPermissions_ShouldReturnNoPermissions_OnInValidPlant()
...
+                .Returns(Task.FromResult(new List<ProcosysProject>()));
+            // Act
+            var result = await _dut.GetAllProjectsAsync(_plant);
+
+            // Assert
+            Assert.AreEqual(0, result.Count);
+        }
+
+        [TestMethod]
+        public async Task GetAllProjectsAsync_ShouldReturnNoProjects_OnInValidPlant()
```
Comment (Cnordbo) [inline src/tests/Equinor.Procosys.Preservation.MainApi.Tests/Permission/MainApiPermissionServiceTests.cs:98]: invalid is "one word" (camelcasing)

**Review by Cnordbo — CHANGES_REQUESTED** [review]

File: src/tests/Equinor.Procosys.Preservation.WebApi.Tests/Misc/ProjectCheckerTests.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        [TestMethod]
+        public async Task EnsureValidProjectAsync_ShouldValidateOK()
+        {
+            // Arrange
+            _permissionCacheMock.Setup(p => p.IsAValidProjectAsync(Plant, _currentUserOid, Project)).Returns(Task.FromResult(true));
+
+            // Act
+            await _dut.EnsureValidProjectAsync(_testRequest);
```
[the remark this answers is not in the document]
Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.WebApi.Tests/Misc/ProjectCheckerTests.cs:46]: How can we assert when it's nothing to Assert on? The test verify that code execute without exception

[the remark this answers is not in the document]
Comment (janid1967) [reply src/tests/Equinor.Procosys.Preservation.WebApi.Tests/Misc/ProjectCheckerTests.cs:46]: Also discussed with @llykkelarsen ... we think test is ok as is ... won't fix

**Review by Cnordbo — APPROVED** [review]

**Pull request merged**